monitoring/grafana/dashboards/CMakeLists.txt

```diff
@@ -10,7 +10,7 @@ if(NOT CEPH_BUILD_VIRTUALENV)
   set(CEPH_BUILD_VIRTUALENV ${CMAKE_BINARY_DIR})
 endif()
 
-if(WITH_GRAFANA)
+if(WITH_GRAFANA AND WITH_GRAFANA_TESTS)
```

Comment:
Or better off check WITH_TEST instead.

Reply:
Guarding this code with `WITH_TESTS` does not work for us because:

1. the `WITH_TESTS` build does not currently download anything from the Internet and we need to run WITH_TESTS builds in the openSUSE Build Service.

Reply:
I think making WITH_GRAFANA_TESTS an option depending on WITH_GRAFANA is reasonable, but I don't know how to do that.

Reply:
sorry, should make this check a dependency of "make check". Namely add it as a test. @ideepika

Reply:
I think what @tchaikov means is: 
```
CMAKE_DEPENDENT_OPTION(WITH_GRAFANA_TESTS "**describe**" ON
  "WITH_TESTS;WITH_GRAFANA" OFF)
```
this shall set grafana tests on, only if `WITH_GRAFANA` and `WITH_TESTS` both are enabled

Reply:
@ideepika That still doesn't work for us, though, because WITH_GRAFANA_TESTS will be set to ON in the WITH_TESTS build.

Reply:
@ideepika Would it be possible to move the `ExternalProject_Add` into the test itself? Then it would only be triggered during "make check", and never during compilation. I guess that's what @tchaikov meant when he wrote "Namely add it as a test."

Reply:
@smithfarm thanks for elaborating, I think we can do something like that by making grafananet-lib depend on target  check instead of target tests: 
https://github.com/ceph/ceph/commit/ce5724effb270b0f57369fd1e9bd82d7cc872676#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR389

Reply:
@ideepika i just realized that the crux of the problem is that cmake does not provide a builtin solution for creating a test which downloads a file, so we cannot use the test as a fixture for another test which extract the downloaded file. what cmake offers is `ExternalProject_Add` and `FetchContent_Declare`, but what they offer is but a target for us. and test **cannot** depend on a target. so we have two options:

1. define a test which uses an external tool like `curl` or `wget` for downloading `grafonnet-lib.0.1.0.tar.gz`. and add fixture for extracting it. see also https://github.com/ceph/ceph/pull/42194#discussion_r682256995
2. as what you are proposing: add the target defined by `ExternalProject_Add` as a dependency of `check`. but please note, this would break the workflow of "make tests; ctest". and forces developers to use "make check" instead.

yes, `WITH_GRAFANA_TESTS` would be an addon. but i am not sure what its use case would be.

Reply:
@ideepika @aaSharma14 i think #42871 is a better fix.